library/core/src/num/fmt.rs

```diff
@@ -31,8 +32,10 @@ impl<'a> Part<'a> {
                     } else {
                         3
                     }
+                } else if v < 10_000 {
+                    4
                 } else {
-                    if v < 10_000 { 4 } else { 5 }
+                    5
                 }
```

Comment:
Remark: and debatable, but IMO this isn't materially better either.

Reply:
This is open-coding a log10 operation, cleaned this up here: #143540

Reply:
@yotamofek good move! I can't wait to see perf results.